homeassistant/auth/__init__.py

```diff
@@ -166,31 +166,35 @@ def get_auth_mfa_module(self, module_id: str) \
     async def async_get_or_create_user(self, credentials: models.Credentials) \
             -> models.User:
         """Get or create a user."""
-        if not credentials.is_new:
-            user = await self.async_get_user_by_credentials(credentials)
-            if user is None:
-                raise ValueError('Unable to find the user.')
-            else:
-                return user
-
         auth_provider = self._async_get_auth_provider(credentials)
-
         if auth_provider is None:
             raise RuntimeError('Credential with unknown provider encountered')
 
-        info = await auth_provider.async_user_meta_for_credentials(
+        meta = await auth_provider.async_user_meta_for_credentials(
             credentials)
 
-        user = await self._store.async_create_user(
-            credentials=credentials,
-            name=info.name,
-            is_active=info.is_active,
-            group_ids=[GROUP_ID_ADMIN],
-        )
+        if credentials.is_new:
+            user = await self._store.async_create_user(
+                credentials=credentials,
+                name=meta.name,
+                is_active=True if meta.is_active is None else meta.is_active,
```

Comment:
See my comments in auth providers. is_active should return by auth provider
```
is_active= meta.is_active
```

If create a new user with is_active=False is not meaningful for some auth provider, the auth provider could choose to reject inactive user login at first place.

user.is_active will be used in the next step of login flow, e.g. generate the refresh_token.

Reply:
Why do you think it's not appropriate to have a default value of ``True`` for ``is_active``? I did want to prevent these redundant checks for ``credentials.is_new`` in each auth provider. Not saying that one provider can't explicitly check for ``credentials.is_new`` and overwrite the default.

I think, ``is_active`` is nothing an auth provider should have to deal with in the first place, for the reasons you mentioned yourself.

Reply:
To say it short, I'd like to have the default policy at one single place, instead of redundant in each auth provider. That approach becomes even more advantageous when group ids can be specified via ``UserMeta`` in the future.

Reply:
Maybe add one line of comment here